pkg/metrics/sketch_series.go

```diff
@@ -78,6 +79,118 @@ func (sl SketchSeriesList) MarshalJSON() ([]byte, error) {
 	return reqBody.Bytes(), err
 }
 
+// MarshalSplitCompress uses the stream compressor to marshal and compress sketch series payloads.
+// If a compressed payload is larger than the max, a new payload will be generated. This method returns a slice of
+// compressed protobuf marshaled gogen.SketchPayload objects. gogen.SketchPayload is not directly marshaled - instead
+// it's contents are marshaled individually, packed with the appropriate protobuf metadata, and compressed in stream.
+// The resulting payloads (when decompressed) are binary equal to the result of marshaling the whole object at once.
+func (sl SketchSeriesList) MarshalSplitCompress() ([]*[]byte, error) {
+	input := bytes.NewBuffer(make([]byte, 0, 1024))
+	output := bytes.NewBuffer(make([]byte, 0, 1024))
```

Comment:
To me it'd make sense to make the caller pass these buffers (+ the `precompressionBuf` maybe) to `MarshalSplitCompress`, so that the caller controls whether to re-use buffers between invocations (for the initial solution, creating new buffers for each invocation might make sense, but let me know your thoughts).

Reply:
This could make sense. However `serializer.SendSketch` [can be called concurrently](https://github.com/DataDog/datadog-agent/blob/6cda68d9561163c1082f4f16ee5aabcea56301a3/pkg/aggregator/aggregator.go#L479) - so we would have to introduce locks to share buffers between sketch serializations. This is essentially the same issue that I ran into with the payload builder optimizations. (However we deemed in that case the locks probably won't hurt performance). So if we are ok with that same assumption here, I can make the change. 

It would be nice to re-structure `serializer`/`aggregator`  to ensure these methods are not called concurrently which would make these sort of optimizations much easier (and cleaner). WDYT?

One other tradeoff is

Reply:
For this PR, I think the caller of `MarshalSplitCompress` can allocate new buffers for each call to `MarshalSplitCompress`, and we can decide in a follow up PR (and/or as part of the payload builder optimizations) how to actually share the buffers with concurrency concerns in mind, WDYT?

Reply:
Sounds good to me.